packages/relay/src/services/PostService.ts

```diff
@@ -1,29 +1,143 @@
+        //         e. publishedAt DESC
+        let statement = `
+            SELECT
+                p.*,
+                CASE
+                    WHEN ${DAY_DIFF_STAEMENT} <= 2 THEN
+                        -0.5 * ${DAY_DIFF_STAEMENT} + p.upCount * 0.2 - p.downCount * 0.2 + p.commentCount * 0.1
+                    ELSE
+                        COALESCE(v.daily_upvotes, 0) + COALESCE(c.daily_comments, 0)
+                END AS sorting_score,
+                v.daily_upvotes AS daily_upvotes,
+                c.daily_comments AS daily_comments
+            FROM
+                Post AS p
+            LEFT JOIN (
+                SELECT
+                    postId,
+                    SUM(CAST(upVote AS INTEGER)) AS daily_upvotes,
+                    SUM(CAST(downVote AS INTEGER)) AS daily_downvotes
+                FROM
+                    Vote
+                WHERE 
+                    ${DAY_DIFF_STAEMENT} = 0
+                GROUP BY
+                    postId
+            ) AS v ON p.postId = v.postId
+            LEFT JOIN (
+                SELECT
+                    postId,
+                    COUNT(*) AS daily_comments
+                FROM
+                    Comment
+                WHERE
+                    ${DAY_DIFF_STAEMENT} = 0
+                GROUP BY
+                    postId
+            ) AS c ON p.postId = c.postId
+            ORDER BY 
+                CASE
+                    WHEN ${DAY_DIFF_STAEMENT} <= 2 THEN 0
+                    ELSE 1
+                END, sorting_score DESC, 
+                CASE
+                    WHEN ${DAY_DIFF_STAEMENT} <= 2 THEN 0
+                    ELSE v.daily_upvotes
+                END DESC,
+                CASE
+                    WHEN ${DAY_DIFF_STAEMENT} <= 2 THEN 0
+                    ELSE COALESCE(c.daily_comments, 0)
+                END DESC,
+                CAST(p.publishedAt AS INTEGER) DESC
+        `
+        // anondb does't provide add column api
+        // use lower level db api directly from postgres / sqlite
+        // for the complex sql statement
+        if (DB_PATH.startsWith('postgres')) {
+            const pg = db as PostgresConnector
+            this.cache = (await pg.db.query(statement)).rows
+        } else {
+            const sq = db as SQLiteConnector
+            this.cache = await sq.db.all(statement)
+        }
+    }
+
+    // returns the LOAD_POST_COUNT posts of the given page
+    // page 1
+    // start = (1 - 1) * LOAD_POST_COUNT = 0 ... start index
+    // end = page + LOAD_POST_COUNT ... end index
+    // slice(page, end) ... the end element will be excluded
     async fetchPosts(
         query: string | undefined,
         epks: string[] | undefined,
+        page: number,
         db: DB
     ): Promise<Post[] | null> {
         if (!query) {
-            const posts = await db.findMany('Post', {
-                where: {
-                    status: 1,
-                },
-            })
```

Comment:
Should we keep above db searching query as fallback value if invalid cache is happening?